modules/libhubbub/0.3.8/patches/inc_files.patch

```diff
@@ -0,0 +1,12393 @@
+From a09aafe9e6b316ebc896e9b9e19b4f53c9af98c5 Mon Sep 17 00:00:00 2001
```

Comment:
Ideally we should not check in such big patch file in BCR, how is this file generated? Can you make it a step in the build process?

Reply:
These files were created by the build tools from this project:
* entities.inc - created using the Perl script build/make-entities.pl. I tried to convert it into a genrule step, but unfortunately this script has hardcoded file paths so it wasn't that easy to do. It looks like I need to fix this Perl script first before it can be used correctly in genrule. The second problem is that Perl is usually available on Linux/MacOS, but is not always available on Windows. Could you give some tips on how I can make perl execution portable across all platforms?
* autogenerated-element-type.inc - created using the GNU gperf utility. There is also the problem that this utility is usually available on Unix platforms, but not on Windows.

So as a first approach I tried this big patch. It's not perfect, but it makes it possible to easily compile a project on multiple platforms.

Reply:
Is Windows a supported system for the library at all? If so, what is official recommended build step? From https://www.netsurf-browser.org/, it sounds like this is mainly for unix systems? If you insist adding Windows support, maybe try [rules_perl](https://registry.bazel.build/modules/rules_perl)?

I'm very uncomfortable on accepting such a big patch, it increases the BCR repo size by a lot, and there is no way to review it properly.

Reply:
https://www.netsurf-browser.org/downloads/windows/

Ok, I will check about rules_perl and will check how properly execute GNU gperf on Windows.

Reply:
Thanks, done!

Reply:
SGTM

Reply:
@meteorcloudy now gperf is run as genrule, so libhubbub patch is now minimal. I removed the draft mark, so PR is marked as ready for review.

Reply:
Thanks! Looks good, just one nit comment!